cpp/src/io/parquet/reader_impl_preprocess.cu

```diff
@@ -659,7 +668,7 @@ struct set_str_dict_index_ptr {
   __device__ void operator()(size_t i)
   {
     auto& chunk = chunks[i];
-    if (chunk.physical_type == BYTE_ARRAY && (chunk.num_dict_pages > 0)) {
+    if (chunk.num_dict_pages > 0 and is_string_chunk(chunk)) {
```

Comment:
nit: is the condition order change intentional? (physical_type check is moved after num_dict_pages check. it's changed at 2 places). not a suggestion, just curious if this order change matters.

Reply:
Yes, because the `is_string_chunk()` call is a bit more expensive, I thought it would be better to be able to short-circuit out and not perform that check if it isn't necessary.